jenkins/run

```diff
@@ -67,26 +85,52 @@ _image=${IMAGE:-${_image_repo}/pbench-${_image_role}-${_image_kind}:${_branch_na
 if [ -n "${CONTAINER_HOST}" ]; then
     # Since CONTAINER_HOST is defined, when we want to run another container, do
     # it with `podman-remote` (which will run it where CONTAINER_HOST points).
-    # This works nicely, e.g., if we are already inside a properly configured
-    # container.
+    # This is the core mechanism that allows CI jobs, which are run inside one
+    # container, to run sub-sets of the build inside separate containers on the
+    # same host.
     PODMAN="podman-remote"
 else
-    # Since CONTAINER_HOST is not defined, don't invoke the container remotely.
-    # However, we might want to run a container "remotely" on this host;
-    # therefore, enable the service which makes it available to receive
-    # `podman-remote` invocations.  This creates a listener on
-    # /run/user/$(id -u)/podman/podman.sock, which will be mapped into the
-    # container in the invocation below so that we can create peer-containers
-    # on this host from inside the container.
+    # Since CONTAINER_HOST is not defined, arrange to invoke the container
+    # directly.  Since the code running in the container might want to run an
+    # additional container "remotely" on this host, enable the service which
+    # makes it available to receive `podman-remote` invocations.  This creates a
+    # listener on /run/user/$(id -u)/podman/podman.sock, which will be mapped
+    # into the container in the invocation below so that we can create
+    # peer-containers on this host from inside the container.
     PODMAN="podman"
     systemctl --user enable --quiet --now podman.socket
 fi
 
+# We have two directory trees which we are concerned with inside the container:
+# $HOME_DIR and $HOME_DIR/pbench.  The first is provided for anything which
+# happens to reference it (such as RPM builds and npm), and the second contains
+# the Git checkout.
+#
+# When this script is run under the CI (or in any other environment which
+# defines $WORKSPACE_TMP), the $HOME_DIR is mapped from $WORKSPACE_TMP.  When
+# this script is run locally by an individual developer, $HOME_DIR is mapped
+# from a Podman volume, to maintain isolation from the surrounding environment;
+# this volume is created automatically by Podman if it does not already exist
+# (it is deleted after the container exits, if it is no longer in use), and
```

Comment:
If "it is deleted after the container exits", why do we have to `podman volume rm` below? Is this comment incorrect, or is there more going on that should be explained?

Reply:
Sadly, I did not find a way to create a temporary but named volume -- a volume can be anonymous, in which case it is deleted implicitly, or named, in which case it must be deleted explicitly.  Since we need to share the volume among the containers, I need to name it, and, therefore, I cannot use a temporary/anonymous volume...and, therefore, it is deleted _by this script_ after the container exits.  (Would adding "by this script" to the text above resolve your concern?)

Reply:
Instead, how about (picking apart the essentials of your explanation) "create a temporary but named volume which will be deleted by this script after the container exits"??